app/models/alchemy/elements_repository.rb

```diff
@@ -76,6 +81,36 @@ def expanded
       self.class.new reject(&:folded)
     end
 
+    # All not nested top level elements
+    # @return [Alchemy::ElementRepository]
+    def not_nested
+      self.class.new(select { |e| e.parent_element_id.nil? })
+    end
+
+    # Elements in reversed order
+    # @return [Alchemy::ElementRepository]
+    def reverse
+      self.class.new elements.reverse
+    end
+
+    # Elements in random order
+    # @return [Alchemy::ElementRepository]
+    def random
+      self.class.new Array(elements).shuffle
+    end
+
+    # Elements off setted by
+    # @return [Alchemy::ElementRepository]
+    def offset(offset)
```

Comment:
I'm not sure we want to support `offset` and `limit`. In the context of a page, there's rarely so many elements we need to paginate, and for something like an index action for all elements site-wide, we probably need to paginate before passing elements into a repository like this. Are these used anywhere?

Reply:
This is used by the [`render_elements` helper ](https://github.com/AlchemyCMS/alchemy_cms/blob/main/lib/alchemy/elements_finder.rb#L47)